MinecraftClient/Program.cs

```diff
@@ -35,59 +39,75 @@ static class Program
 
         /// <summary>
         /// The main entry point of Minecraft Console Client
         /// </summary>
-        static void Main(string[] args)
-        {
-            Console.WriteLine("Console Client for MC {0} to {1} - v{2} - By ORelio & Contributors", MCLowestVersion, MCHighestVersion, Version);
-
-            //Build information to facilitate processing of bug reports
-            if (BuildInfo != null)
-            {
-                ConsoleIO.WriteLineFormatted("§8" + BuildInfo);
-            }
-
-            //Debug input ?
-            if (args.Length == 1 && args[0] == "--keyboard-debug")
-            {
-                ConsoleIO.WriteLine("Keyboard debug mode: Press any key to display info");
-                ConsoleIO.DebugReadInput();
+        static void Main(string[] args) {
+            if (args.Length > 0) {
+                if (args.Contains("--help")) {
+                    Console.WriteLine("Command-Line Help:");
+                    Console.WriteLine("MinecraftClient.exe <username> <password> <server>");
+                    Console.WriteLine("MinecraftClient.exe <username> <password> <server> \"/mycommand\"");
+                    Console.WriteLine("MinecraftClient.exe --setting=value [--other settings]");
+                    Console.WriteLine("MinecraftClient.exe --section.setting=value [--other settings]");
+                    Console.WriteLine("MinecraftClient.exe <settings-file.ini> [--other settings]");
+                    return;
+                }
             }
 
             //Setup ConsoleIO
             ConsoleIO.LogPrefix = "§8[MCC] ";
-            if (args.Length >= 1 && args[args.Length - 1] == "BasicIO" || args.Length >= 1 && args[args.Length - 1] == "BasicIO-NoColor")
-            {
-                if (args.Length >= 1 && args[args.Length - 1] == "BasicIO-NoColor")
-                {
+            if (args.Length >= 1 && args[args.Length - 1] == "BasicIO" ||
+                args.Length >= 1 && args[args.Length - 1] == "BasicIO-NoColor") {
+                if (args.Length >= 1 && args[args.Length - 1] == "BasicIO-NoColor") {
                     ConsoleIO.BasicIO_NoColor = true;
                 }
+
                 ConsoleIO.BasicIO = true;
                 args = args.Where(o => !Object.ReferenceEquals(o, args[args.Length - 1])).ToArray();
             }
 
-            //Take advantage of Windows 10 / Mac / Linux UTF-8 console
-            if (isUsingMono || WindowsVersion.WinMajorVersion >= 10)
-            {
-                Console.OutputEncoding = Console.InputEncoding = Encoding.UTF8;
+            if (ConsoleIO.BasicIO) {
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+                    Console.WriteLine("BasicIO is not supported on non-Windows systems.");
+                    Console.WriteLine("Press any key to continue...");
+                    Console.ReadKey();
+                    Environment.Exit(1);
+                }

```

Comment:
Any reason to prevent BasicIO on Linux? The BasicIO mechanism should rely on ReadLine() and WriteLine() only without console formatting or manipulating the terminal, so that should work on all OS 🤔
... Or is that related to the lock in System.Console you talked about before?

Reply:
Yes. We can't `Console.ReadLine` and `Console.WriteLine` at the same time on non-windows systems because of the lock.
Sorry, my wording wasn't really clear about that :P